cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java

```diff
@@ -154,11 +211,21 @@ private void startExecution(LoadMonitor loadMonitor) {
     }
     _hasOngoingExecution = true;
     _stopRequested.set(false);
+      if (_isKafkaAssignerMode) {
+        _numExecutionStartedInKafkaAssignerMode.incrementAndGet();
+      } else {
+        _numExecutionStartedInNonKafkaAssignerMode.incrementAndGet();
+      }
     _proposalExecutor.submit(new ProposalExecutionRunnable(loadMonitor));
   }
 
-  public synchronized void stopExecution() {
-    _stopRequested.set(true);
+  public synchronized void stopExecution(boolean isOnPurpose) {
```

Comment:
I suspect that the proposed change would potentially cause miscounting the counter/metric in case there are multiple on-purpose stop requests. The existing function guarantees that only when an actual stop is triggered, the relevant counter(s) will increase.

Reply:
How about using `boolean stopExecution()`? This method returns `true` if `_stopRequested` flag is set otherwise it returns `false`. And then in `boolean useTriggeredStopExecution()` increment the counter based on return value `stopExecution()`